Projects/BookJSON2/main.go

```diff
@@ -30,26 +31,39 @@ var books []Book
 func (b *Books) book(w http.ResponseWriter, r *http.Request) {
 	// if r.Method == POST
 	switch r.Method {
-	case "POST": //Retriving a book with Id number
+	case "GET": //Retriving a book with Id number
 		{
+
 			w.Header().Set("content-type", "application/json")
-			jsonreqestfromweb := &Book{}
-			err := json.NewDecoder(r.Body).Decode(&jsonreqestfromweb)
+			val := r.FormValue("id")
+			id, err := strconv.Atoi(val)
 			if err != nil {
-				fmt.Println(err)
-				return
+				panic(err)
+			}
+			if id >= len(b.Books) {
+				w.WriteHeader(http.StatusNotFound)
+
+			} else {
```

Comment:
Usually go is written in such a way that each blocks that means an end, returns. Sometimes that simplifies nesting (else block not needed).

```suggestion
			if id >= len(b.Books) {
				w.WriteHeader(http.StatusNotFound)
				return
			}
```

Reply:
resolved with commit https://github.com/AmanSarraf/Go-Project/pull/9/commits/85fdee5b41960dddf4790f50c6def83cb4fe3ea5